python/lsst/ts/salobj/configurable_csc.py

```diff
@@ -0,0 +1,376 @@
+
+        Parameters
+        ----------
+        id_data : `CommandIdData`
+            Command ID and data
+
+        Notes
+        -----
+        The ``settingsToApply`` field must be one of:
+
+        * The name of a config label or config file
+        * The name and version of a config file, formatted as
+          ``<file_name>:<version>``, where the version is a git reference,
+          such as a git tag or commit hash. This form does not support labels.
+        """
+        config_name = id_data.data.settingsToApply
+        if config_name:
+            name_version = config_name.split(":")
+            if len(name_version) == 2:
+                config_file_name, githash = name_version
+                config_file_path = self.config_dir / config_file_name
+                try:
+                    config_yaml = subprocess.check_output(["git", "cat-file", "--textconv",
+                                                          f"{githash}:{config_file_path}"],
+                                                          stderr=subprocess.STDOUT)
+                except subprocess.CalledProcessError as e:
+                    raise base.ExpectedError(f"Could not read config {config_name}: {e.output}")
+            elif len(name_version) == 1:
+                config_file_name = self.config_label_dict.get(config_name, config_name)
+                config_file_path = self.config_dir / config_file_name
+                if not config_file_path.is_file():
+                    raise base.ExpectedError(f"Cannot find config file {config_file_name} "
+                                             f"in {self.config_dir}")
+                with open(config_file_path, "r") as f:
+                    config_yaml = f.read()
+            else:
+                raise base.ExpectedError(f"Could not parse {config_name} as name or name:version")
+
+            user_config_dict = yaml.safe_load(config_yaml)
+        else:
+            user_config_dict = {}
+        try:
+            full_config_dict = self.config_validator.validate(user_config_dict)
+        except Exception as e:
+            raise base.ExpectedError(f"schema {config_file_path} failed validation: {e}")
+        config = types.SimpleNamespace(**full_config_dict)
+        self.configure(config)
+        self.evt_appliedSettingsMatchStart.set_put(appliedSettingsMatchStartIsTrue=True, force_output=True)
+
+    @abc.abstractmethod
+    def configure(self, config):
+        """Configure the CSC.
+
+        Parameters
+        ----------
+        config : `object`
+            The configuration as described by the schema at ``schema_path``,
+            as a struct-like object.
+
+        Notes
+        -----
+        Called when running the ``start`` command, just before changing
+        summary state from `State.STANDBY` to `State.DISABLED`.
+        """
+        raise NotImplementedError("Subclasses must override")
+
+    @classmethod
```

Comment:
It seems to me that this could be a `staticmethod` instead.

Reply:
Yes, I thought about that. I felt it a bit safer to leave it a classmethod in case the code wanted to somehow auto-generate it based on class information. But I freely admit that is unlikely and will happily change it if you wish.

Reply:
Yes, I thought about that as well. I think a `staticmethod` will pass a better message, which is "don't try to be clever about this". :)

Reply:
OK. I have changed it (as new commits)